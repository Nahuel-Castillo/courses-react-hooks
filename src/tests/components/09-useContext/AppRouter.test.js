import { mount } from 'enzyme';
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Test of <AppRouter/>', () => {
    
    const user = {
        id: 124,
        name: 'Nahuel'
    };

    const wrapper = mount(
        <UserContext.Provider value={ user }>
            <AppRouter/>
        </UserContext.Provider>
    );

    test('should show correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });
});