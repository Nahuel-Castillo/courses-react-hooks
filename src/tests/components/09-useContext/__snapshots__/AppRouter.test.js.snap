// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <AppRouter/> should show correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        UseContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h2>
         HomeScreen 
      </h2>
      <hr />
      <pre />
    </div>
  </div>
</div>
`;
